tests/rptest/tests/data_transforms_test.py

```diff
@@ -433,3 +442,234 @@ def test_leadership_changing_randomly(self):
+
+        EXPECTED_ATTRS = [
+            Attribute({
+                'key': 'transform_name',
+                'value': {
+                    'stringValue': 'any'  # don't care for validation
+                }
+            }),
+            Attribute({
+                'key': 'node',
+                'value': {
+                    'intValue': 23  # don't care for validation
+                }
+            })
+        ]
+
+        BODY_FIELD = 'body'
+        TS_FIELD = 'timeUnixNano'
+        SEVERITY_FIELD = 'severityNumber'
+        ATTRS_FIELD = 'attributes'
+
+        def __init__(self, raw: str):
+            self._record = json.loads(raw)
+            self._record['value'] = json.loads(self._record['value'])
+
+            # Redpanda bits
+            self.offset = self._record['offset']
+            self.key = self._record['key']
+
+            # OTel bits
+            self._rep = self._record['value']
+            self.body = self._rep.get(self.BODY_FIELD, None)
+            self.timestamp_ns = self._rep.get(self.TS_FIELD, None)
+            self.severity = self._rep.get(self.SEVERITY_FIELD, None)
+            self.attributes = [
+                self.Attribute(attr)
+                for attr in self._rep.get(self.ATTRS_FIELD, [])
+            ]
+
+        def __str__(self):
+            return json.dumps(self._record, indent=1)
+
+        def validate(
+                self,
+                offset: typing.Optional[int] = None
+        ) -> typing.Optional[list[str]]:
+            errors = []
+            if offset is not None and offset != self.offset:
+                errors.append(f"Expected offset {offset} got {self.offset}")
+            if self.body is None:
+                errors.append(f"Missing {self.BODY_FIELD}")
+            if self.timestamp_ns is None:
+                errors.append(f"Missing {self.TS_FIELD}")
+            if self.severity is None:
+                errors.append(f"Missing {self.SEVERITY_FIELD}")
+            if self.ATTRS_FIELD not in self._rep:
+                errors.append(f"Missing {self.ATTRS_FIELD}")
+
+            if len(self.attributes) != 2:
+                errors.append(
+                    f"Expected 2 attributes, got {len(self.attributes)}")
+
+            for attr in self.EXPECTED_ATTRS:
+                if attr not in self.attributes:
+                    errors.append(f"Missing attribute '{attr.key}'")
+
+            if len(errors) > 0:
+                return errors
+
+            return None
+
+    def setup_identity_xform(self):
+        it, ot = self.topics
+        self._deploy_wasm(name="identity-logging-xform",
+                          input_topic=self.topics[0],
+                          output_topic=self.topics[1],
+                          file="tinygo/identity_logging.wasm")
+        return self.topics
+
+    def consume_one_log_record(self, offset=0, timeout=10) -> LogRecord:
+        return self.LogRecord(
+            self._rpk.consume(self.logs_topic.name,
```

Comment:
Question: do all these `rpk` commands need to be wrapped in retries?

Reply:
I don't really think so. The timeout gets applied to the rpk shell-out under the hood of `RpkTool`, and consume should just sit there until we either receive a record back OR we kill the subprocess. In the latter case we get a nice RpkException. Did you have a particular failure mode in mind?

Reply:
No just wanted to make sure these tests are robust as possible. For Admin requests that usually means wrapping stuff in retries.

Reply:
Yeah, it's a fair point. If rpk fails to establish a connection that'll immediately fail the test.

Reply:
It's probably fine, we can see if any issues crop up first